frontend/src/core/comments/components/AddComment.js

```diff
@@ -25,69 +41,212 @@ export default function AddComments(props: Props) {
+    const [target, setTarget] = React.useState();
+    const [index, setIndex] = React.useState(0);
+    const [search, setSearch] = React.useState('');
+    const renderElement = React.useCallback(props => <Element {...props} />, []);
+    const editor = React.useMemo(
+        () => withMentions(withReact(createEditor())), []
+    );
+
+    const chars = USERS.filter(c =>
+        c.toLowerCase().startsWith(search.toLowerCase())
+    ).slice(0, 10);
+
+    React.useEffect(() => {
+        if (target && chars.length > 0) {
+            disableHistoryScroll(true);
+            const el = ref.current;
+            const domRange = ReactEditor.toDOMRange(editor, target);
+            const rect = domRange.getBoundingClientRect();
+            const suggestionsHeight = el.clientHeight + 10;
+            rect.top + window.pageYOffset + (suggestionsHeight) > window.innerHeight 
+            ? el.style.top = `${rect.top + window.pageYOffset - (suggestionsHeight)}px`
+            : el.style.top = `${rect.top + window.pageYOffset + 24}px`;
+            el.style.left = `${rect.left + window.pageXOffset}px`;
+        }
+    }, [chars.length, editor, index, search, target]);
+
+    const mentionsKeyDown = React.useCallback(
+        event => {
+            if (target) {
+                switch (event.key) {
+                    case 'ArrowDown': {
+                        event.preventDefault();
+                        const prevIndex = index >= chars.length - 1 ? 0 : index + 1;
+                        setIndex(prevIndex);
+                        break;
+                    }
+                    case 'ArrowUp': {
+                        event.preventDefault();
+                        const nextIndex = index <= 0 ? chars.length - 1 : index - 1;
+                        setIndex(nextIndex);
+                        break;
+                    }
+                    case 'Tab':
+                    case 'Enter':
+                        event.preventDefault();
+                        Transforms.select(editor, target);
+                        insertMention(editor, chars[index]);
+                        disableHistoryScroll(false);
+                        setTarget(null);
+                        break;
+                    case 'Escape':
+                        event.preventDefault();
+                        disableHistoryScroll(false);
+                        setTarget(null);
+                        break;
+                    default:
+                        return;
+                }
+            }
+        },
+        [chars, editor, index, target]
+    );
+
+    const onKeyDown = (event: SyntheticKeyboardEvent<>) => {
+        if (event.key === 'Enter' && !event.shiftKey) {
+            event.preventDefault();
+            submitComment();
+        }
+        if (event.key === 'Enter' && event.shiftKey) { 
+            event.preventDefault();
+            editor.insertText("\n\uFEFF");
```

Comment:
This allows for the new lines to render while adding comments. The good news is that it might not interfere with how `\n` have been handled in current comments.

To avoid an issue with the cursor placement and an error when navigating with arrows that occurs in Firefox I couldn't have `\n` be the last character so I added the BOM as per the recommendation given in Slate's Slack channel. 

I'm still working on how to strip this when it is the only thing on a line as it still renders a blank line (or an entire blank comment) ATM.

Reply:
Nice trick! Please also add (most of) this comment as the code comment.